src/SonarScanner.Shim/PropertiesWriter.cs

```diff
@@ -284,6 +284,27 @@ private static bool IsAscii(char c)
             return c <= sbyte.MaxValue;
         }
 
+        internal /* for testing purposes */ string JoinAsSonarQubeMultiValueProperty(IEnumerable<string> paths)
```

Comment:
Minor: how about _EncodeAsSonarQubeMultiValueProperty_ as a name?

Reply:
Yep! That's better!